.github/workflows/dev-test.yml

```diff
@@ -0,0 +1,84 @@
+# Run tests against developer push
+
+# Runs jobs on linux and windows to maintain compatibility.
+# The main repo is checked out into a sub-directory because an optional
+# second repo may be checked out for testing.
+
+name: dev-test
+
+on:
+  push:
+    branches-ignore: [ master, main ]
+
+jobs:
+  build-test:
+    name: Node v${{ matrix.node-version }} on ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version: [14.x]
+        os: [ubuntu-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+
+    # job level env vars used for a couple of steps
+    env:
+      # see github-glue.test.ts for info on setting these
+      GGG_TOKEN: ${{ secrets.GGG_TOKEN }}
+      GGG_REPOSITORY: ${{ secrets.GGG_REPOSITORY }}
+
+    steps:
+    # Check out repo under sub-dir so other repo can be checked out
+    - uses: actions/checkout@v2
+      with:
+        path: gitgitgadget
+
+    # Check out github-glue.test.ts repo if configured for it
+    - uses: actions/checkout@v2
+      with:
+        repository: ${{ github.repository_owner }}/${{ env.GGG_REPOSITORY }}
+        token: ${{ env.GGG_TOKEN }}
+        path: ${{ env.GGG_REPOSITORY }}
+      if: env.GGG_TOKEN
+
+    - uses: actions/setup-node@v1.4.2
+      with:
+        node-version: '14.x' # optional, default is 10.x
+
+    - name: Install packages
+      run: npm ci
+      working-directory: gitgitgadget
+
+    - name: Set git smtp config
+      shell: bash
+      env:
+        # see ci-helper.test.ts for info on setting these
+        SMTPUSER: ${{ secrets.SMTPUSER }}
+        SMTPPASS: ${{ secrets.SMTPPASS }}
+        SMTPHOST: ${{ secrets.SMTPHOST }}
+        SMTPOPTS: ${{ secrets.SMTPOPTS }}
+      run: |
+        git config --add gitgitgadget.cismtpuser $SMTPUSER
+        git config --add gitgitgadget.cismtphost $SMTPHOST
+        git config --add gitgitgadget.cismtpopts "$SMTPOPTS"
+        git config --add gitgitgadget.cismtppass $SMTPPASS
+      if: env.SMTPUSER
+      working-directory: gitgitgadget
```

Comment:
I wonder whether it might make more sense to combine these into a value ready for `GIT_CONFIG_PARAMETERS` (which is the environment variable that backs the `git -c <var>=<value>` feature: try for yourself in a Bash: `GIT_CONFIG_PARAMETERS="'chris.handle=webster'" git config chris.handle`). It could then be used _just_ in the "Run tests" job below, in the `env` section:

```yaml
    env:
        GIT_CONFIG_PARAMETERS: ${{ secrets.GGG_SMTP_CONFIG_PARAMETERS }}
```

The secret would then look somewhat like this:

```
'gitgitgadget.cismtpuser=hello-world' 'gitgitgadget.cismtphost=mail.example.com' 'gitgitgadget.cismtppass=xoxo'
```

What do you think?

Reply:
It works but updating secrets is a replace and not an edit so updating the password would be a real pain.  How about:

```
    env:
        GIT_CONFIG_PARAMETERS: ${{ secrets.GGG_SMTP_USER secrets.GGG_SMTP_PASS secrets.GGG_SMTP_HOST secrets.GGG_SMTP_OPTS }}
```
Then each secret would be in the format you you showed.  This would not prevent you from stuffing everything into one secret and not setting the other three.

Maybe the `Set git test repo config` step can just set a var to use here as well.
```
echo "::set-env name=GGG_REPO::'gitgitgadget.githubtest.githubuser=$GITHUB_REPOSITORY_OWNER' 'gitgitgadget.$GITHUB_REPOSITORY_OWNER.githubrepo=$GGG_REPOSITORY' 'gitgitgadget.$GITHUB_REPOSITORY_OWNER.githubtoken=$GGG_TOKEN'"
```
This avoids running `git config` but still requires a step.  Your thoughts?

Reply:
Between all the options we have on the table so far, I prefer the "secret `GIT_CONFIG_PARAMETERS` and your current solution as a close runner-up. Since the latter does not require any additional work from you, let's just keep it as-is.